File: dalme_api/serializers/locales.py
from dalme_app.models import LocaleReference

from .base_classes import DynamicSerializer
from .countries import CountryReferenceSerializer


class LocaleReferenceSerializer(DynamicSerializer):
    """Serializer for locales."""

    country = CountryReferenceSerializer(field_set='attribute')

    class Meta:  # noqa: D106
        model = LocaleReference
        fields = ('id', 'name', 'administrative_region', 'country', 'latitude', 'longitude')
        field_sets = {
            'option': ['id', 'name', 'country'],
            'attribute': ['id', 'name', 'administrative_region', 'country'],
        }

File: ui/src/schemas/locales.js
import * as yup from "yup";

export const localeSchema = yup
  .object()
  .shape({
    id: yup.number().required(),
    name: yup.string().required(),
    administrativeRegion: yup.string().required(),
    country: yup
      .object()
      .shape({
        id: yup.number().required(),
        name: yup.string().required(),
      })
      .required(),
    latitude: yup.number().required(),
    longitude: yup.number().required(),
  })
  .camelCase();

export const localeListSchema = yup.array().of(localeSchema);
